### PR TITLE
Tempo: Allow quotes in tag names and attributes

### LINKED ISSUE #77394: [TraceQL] Support quoted attribute names
Tempo now has the ability to support quoted attributes. This allows users to query attribute names that have spaces and other esoteric characters.

This is now a valid way to query the `span.db.statement` field:

```
{ span."db.statement" != "" }
```

There are two escape sequences `\"` and `\\` for obvious reasons.

Currently Grafana just highlights this as a string and does not correctly show autocomplete:

image

Quotes can be started and ended at any time inside of the attribute string so this is technically valid:
```
{ span.db.st"ateme"nt != "" }
```

### PR DESCRIPTION
**Pull request by fabrizio-grafana**
Allow tag names and values to use `"`. Examples:
image
image

Note that this allows to write queries that might look wrong at first sight. Example (note the missing closing double quote after `GET`): 
image

Also, note that for autocompletion to work properly, we need a corresponding fix in Tempo, [as mentioned here](https://github.com/grafana/grafana/issues/77394#issuecomment-1794797114).

Fixes https://github.com/grafana/grafana/issues/77394
Fixes https://github.com/grafana/grafana/issues/77685

### CODE REVIEW

**Comment by joe-elliott** [comment]
Yup, this definitely makes some horrible things possible :). This is fine because if you are naming your tags:

`he\ll o world  $ =    GET`

Then it's ok that your TraceQL queries look unreadable.

**Review by joey-grafana — COMMENTED** [review]
Could we get a more detailed error message for this? 

Screenshot 2023-11-15 at 20 31 16



Also, honestly I'm not sure why we need to support some of these strange strings with quotes all over the place :D Is it because the quotes can technically be anywhere in the string and we are not sure of all the cases where it would be invalid/valid so it would be difficult to restrict where the quotes can be? (so we just basically allow the quotes in many places)

File: public/app/plugins/datasource/tempo/traceql/traceql.ts [hunk 1]
```diff
@@ -75,23 +75,29 @@ export const language: languages.IMonarchLanguage = {
 
       // keywords
       [
+        // match only prefefined keywords
```
Comment (joey-grafana) [inline public/app/plugins/datasource/tempo/traceql/traceql.ts:78]: typo

**Review by joey-grafana — APPROVED** [review]
LGTM 👍

**Pull request merged**